pkg/server/server.go

```diff
@@ -202,16 +290,28 @@ func (s *ProxyServer) getFrontendsForBackendConn(agentID string, backend Backend
 }
 
 // NewProxyServer creates a new ProxyServer instance
-func NewProxyServer(serverID string, serverCount int, agentAuthenticationOptions *AgentTokenAuthenticationOptions) *ProxyServer {
-	bm := NewDefaultBackendManager()
+func NewProxyServer(serverID string, proxyStrategies []ProxyStrategy, serverCount int, agentAuthenticationOptions *AgentTokenAuthenticationOptions) *ProxyServer {
+	var bms []BackendManager
+	for _, ps := range proxyStrategies {
+		switch ps {
+		case ProxyStrategyDestHost:
+			bms = append(bms, NewDestHostBackendManager())
+		default:
+		}
+	}
+	bms = append(bms, NewDefaultBackendManager())
```

Comment:
Seems like the default behavior (i.e. no proxy strategies provided) should be to just have the default. However if you *do* list strategies I think it is legitimate to leave out the default manager. Essentially saying its not ok to attempt to use a random backend.

Reply:
I actually like the original cascading behavior where a fallback (the default manager) is provided if the user selected manager fails in some way. I don't think including this should cause any problems, unless if in the case that the selected manager fails, we want to fail the routing instead of finding a different path?

Reply:
I would agree that if no strategy is provided that using the default is correct. I would also agree that most users would want the default as a fallback. However that could be accomplished by setting (for example) "--proxy-strategies=destHost,default". This issue is that is nodes do not have inter-communication or worse yet have it but its not desirable, then you cannot avoid it. If we allow default to be disabled (eg. by setting "--proxy-strategies=destHost") then whoever is provisioning the cluster can choose the right behavior for them.

Reply:
I agree that we should allow users to disable the default proxy strategy if they know exactly what proxy strategy they wanna use. 

What about if no strategies are provided, i.e., "--proxy-strategies" is not specified, the default strategy will be adopted. But if the user specifies any proxy strategies without including the default strategy, e.g. "--proxy-strategies=destHost", then we won't fall back to default, if no backends are available.

Reply:
Ahh thanks for the explanation. +1 from me. This is an elegant way for users to decide if they want to incorporate fallback strategies or not.

Reply:
So I think we are all in agreement about the desired behavior. However I think to implement that we would need to add a strategy name for the default backend manager. (I'd be fine with "default"). Then this addition would occur if that was specified not in all cases as this code shows.